src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CommitManager.cs

```diff
@@ -316,12 +316,6 @@ internal static bool IsCommitCharacter(CompletionRules completionRules, Completi
                 }
             }
 
-            // general rule: if the filtering text exactly matches the start of the item then it must be a filter character
-            if (Helpers.TextTypedSoFarMatchesItem(item, textTypedSoFar))
-            {
-                return false;
-            }
```

Comment:
Why was this removed?  it seems like something intentional.

Reply:
There's two main reasons:
- The check doesn't seem to affect anything. After running all the completion tests, the only test that failed when the check was removed was a test that [tested an impossible scenario](https://github.com/dotnet/roslyn/pull/45099/files#diff-6924c1ebe069be209b3f9f77c73d9e13R354).
- We are hoping to use [similar logic](https://github.com/dotnet/roslyn/pull/45099/files#diff-0735f44f689a499a4e9ecb6aa4a5c2deR120) for LSP commit characters. LSP completion doesn't support text typed so far (and I don't believe there are plans to do so). So I think making the logic as similar as possible would be ideal.

Reply:
can you validate some typing scenarios that involve items in teh list that have characters that aren't identifier chars?  i.e. making sure that `@` and `:` and `=` and ` ` are handled properly for c#?  (and similar for VB).

`@` appears for identifiers that conflict with keyword names.  i.e. `@int`
`:` appears for named args
` = ` appears for attribute properties.

Reply:
It looks like you're right and some of those scenarios currently don't work. :(

- `@`: committing an item that starts with `@` can result in duplicate `@@` - issue filed: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1142478
- `:` testing `:` is currently impossible due to this bug which crashes LSP: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1132899/
- `=`: wasn't working properly before, but just pushed a fix.

I appreciate the validation suggestions!

Reply:
For all the cases, we should have tests btw :)

Reply:
Note: i would consider these issues current blockers on this as they break muscle memory.

Reply:
`:` crashes LSP...????????????????????? :)